src/ts/ConvivaAnalyticsTracker.ts

```diff
@@ -143,6 +154,52 @@ export class ConvivaAnalyticsTracker {
     return !this._isAdBreakActive;
   }
 
+  public canAttachPlayer(player: PlayerAPI) {
+    let reason: string | undefined = undefined;
+
+    if (this.isPlayerAttached) {
+      reason = 'Player is already attached';
+    }
+
+    if (player.getSource()) {
+      reason = 'Player.load() has already been called (attaching player is possible only before player.load())';
+    }
+
+    return {
+      canAttach: !reason,
+      reason,
+    };
+  }
+
+  public attachPlayer(player: PlayerAPI): void {
+    const {canAttach} = this.canAttachPlayer(player);
+
+    if (!canAttach) {
+      return;
+    }
```

Comment:
This way the relation is more explicit, otherwise you would untangle the implicit link in your head and memorize it, which increases complexity IMO. There is actually a pattern "can execute / execute" I used here.